packages/backend/src/models/SavedChartModel.ts

```diff
@@ -963,6 +964,115 @@ export class SavedChartModel {
+                customMetrics: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT (sqvam.table || '_' || sqvam.name)) FILTER (WHERE sqvam.name IS NOT NULL), '{}')",
+                ),
+                customMetricsBaseDimensions: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT (sqvam.table || '_' || sqvam.base_dimension_name)) FILTER (WHERE sqvam.base_dimension_name IS NOT NULL), '{}')",
+                ),
+                customBinDimensions: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT sqvcd.id) FILTER (WHERE sqvcd.id IS NOT NULL), '{}')",
+                ),
+                customSqlDimensions: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT sqvcsd.id) FILTER (WHERE sqvcsd.id IS NOT NULL), '{}')",
+                ),
+                sorts: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT sqvs.field_name) FILTER (WHERE sqvs.field_name IS NOT NULL), '{}')",
+                ),
+            })
+            .leftJoin(
+                SpaceTableName,
+                'saved_queries.space_id',
+                'spaces.space_id',
+            )
+            .leftJoin(
+                ProjectTableName,
+                'spaces.project_id',
+                'projects.project_id',
+            )
+            .leftJoin(
+                SavedChartVersionsTableName,
+                'saved_queries.saved_query_id',
+                'saved_queries_versions.saved_query_id',
+            )
+            .leftJoin(
+                'saved_queries_version_fields as svf',
+                'saved_queries_versions.saved_queries_version_id',
+                'svf.saved_queries_version_id',
+            )
+            .leftJoin(
+                'saved_queries_version_table_calculations as sqvtc',
+                'saved_queries_versions.saved_queries_version_id',
+                'sqvtc.saved_queries_version_id',
+            )
+            .leftJoin(
+                'saved_queries_version_additional_metrics as sqvam',
+                'saved_queries_versions.saved_queries_version_id',
+                'sqvam.saved_queries_version_id',
+            )
+            .leftJoin(
+                'saved_queries_version_custom_dimensions as sqvcd',
+                'saved_queries_versions.saved_queries_version_id',
+                'sqvcd.saved_queries_version_id',
+            )
+            .leftJoin(
+                'saved_queries_version_custom_sql_dimensions as sqvcsd',
+                'saved_queries_versions.saved_queries_version_id',
+                'sqvcsd.saved_queries_version_id',
+            )
+            .leftJoin(
+                'saved_queries_version_sorts as sqvs',
+                'saved_queries_versions.saved_queries_version_id',
+                'sqvs.saved_queries_version_id',
+            )
+            .where(
+                'saved_queries_versions.saved_queries_version_id',
+                this.database('saved_queries_versions')
```

Comment:
this subselect can be the new bottleneck, but needed to the the latest version. let's keep eye.